src/components/scs/ScSheet/ScSheet.js

```diff
@@ -53,55 +89,199 @@ const ScSheet = ({ sc, classes, intl }) => {
       values[i][propKey] = event.target.value;
-      setScProjectFields(values);
+      setProjectEmployeeFields(values);
     }
   };
 
-  const addFields = type => {
-    const initialFieldsData = {
-      headline: '',
-      weight: 0,
-      percentage: 0,
-      evaluation: undefined,
-      description: '',
-      goal: '',
-      goalComment: ''
-    };
+  const handleChangePropKeyReviewer = (type, i, propKey, event) => {
     if (type === 'dailyBusiness') {
-      const values = [...scDailyBusinessFields];
+      const values = [...dailyBusinessReviewerFields];
+      values[i][propKey] = event.target.value;
+      setDailyBusinessReviewerFields(values);
+    } else {
+      const values = [...projectReviewerFields];
+      values[i][propKey] = event.target.value;
+      setProjectReviewerFields(values);
+    }
+  };
+
+  const addFieldsEmployee = type => {
+    if (type === 'dailyBusiness') {
+      const values = [...dailyBusinessEmployeeFields];
       values.push(initialFieldsData);
-      setScDailyBusinessFields(values);
+      setDailyBusinessEmployeeFields(values);
     } else {
-      const values = [...scProjectFields];
+      const values = [...projectEmployeeFields];
       values.push(initialFieldsData);
-      setScProjectFields(values);
+      setProjectEmployeeFields(values);
     }
   };
 
-  const removeFields = (type, i) => {
+  const addFieldsReviewer = type => {
     if (type === 'dailyBusiness') {
-      const values = [...scDailyBusinessFields];
+      const values = [...dailyBusinessReviewerFields];
+      values.push(initialFieldsData);
+      setDailyBusinessReviewerFields(values);
+    } else {
+      const values = [...projectReviewerFields];
+      values.push(initialFieldsData);
+      setProjectReviewerFields(values);
+    }
+  };
+
+  const removeFieldsEmployee = (type, i) => {
+    if (type === 'dailyBusiness') {
+      const values = [...dailyBusinessEmployeeFields];
+      values.splice(i, 1);
+      setDailyBusinessEmployeeFields(values);
+    } else {
+      const values = [...projectEmployeeFields];
+      values.splice(i, 1);
+      setProjectEmployeeFields(values);
+    }
+  };
+
+  const removeFieldsReviewer = (type, i) => {
+    if (type === 'dailyBusiness') {
+      const values = [...dailyBusinessReviewerFields];
       values.splice(i, 1);
-      setScDailyBusinessFields(values);
+      setDailyBusinessReviewerFields(values);
     } else {
-      const values = [...scProjectFields];
+      const values = [...projectReviewerFields];
       values.splice(i, 1);
-      setScProjectFields(values);
+      setProjectReviewerFields(values);
     }
   };
 
+  const disableSaveButton = () => {
+    if (user.isReviewerInSc(sc)) {
+      return sc.statusSet.includes('REVIEWER_SUBMITTED');
+    }
+
+    if (user.isOwnerInSc(sc)) {
+      return sc.statusSet.includes('EMPLOYEE_SUBMITTED');
+    }
+
+    return true;
+  };
+
+  const renderScFields = user => {
```

Comment:
This could be a separated component, where 'user' would be received as props. We should avoid massive length of components for better maintainability.